torch/_tensor_str.py

```diff
@@ -293,6 +297,14 @@ def _str_intern(inp):
     # within this function.
     # TODO(albanD) This needs to be updated when more than one level is supported
     self, tangent = torch.autograd.forward_ad.unpack_dual(inp)
+    # print shape of the tensor
```

Comment:
This is good but would you elaborate in this comment on why we're excluding these other types and why we've chosen "shape" unlike "size" used by some of them?

Reply:
Added some comments. @mruberry check once if that is ok?